Lib/fontTools/designspaceLib/__init__.py

```diff
@@ -1099,10 +1100,10 @@ def fromstring(cls, string, readerClass=None, writerClass=None):
         return self
 
     def tostring(self, encoding=None):
-        if encoding is unicode or (
+        if encoding is str or (
```

Comment:
This is such a weird line, I checked it before and I still don’t understand how it was supposed to work. Did callers actually pass Python 2 `unicode` function as the encoding?

Reply:
Looks like that's how it was written.

Reply:
Shall I just remove the `encoding is str` bit? It was already weird for Python 2, and now it's even weirder.

Btw. the same thing is written in `fontTools.misc.etree`:

https://github.com/fonttools/fonttools/blob/a363bd75a19553ff5449a05fb7725d913232c36a/Lib/fontTools/misc/etree.py#L153

This should be changed in the same way. Any objections to removing it?

Reply:
That one in etree is weirder than I thought: it mimics the lxml etree API when we don't have lxml, but lxml's `ElementTree.write` doesn't support such a value for the "encoding" argument.

The case in designspaceLib may be caused by looking at the fontTools.misc.etree api, because that's where that call will eventually go to. Perhaps @anthrotype remembers? He implemented `DesignSpaceDocument().tostring()`.

Either way, I strongly suggest to get rid of the `encoding is unicode`/`encoding is str` cases.

Reply:
Even more: an encoding value of `"unicode"` does not work for `DesignSpaceDocument().tostring()` when lxml is installed. So either some fixing needs to be done, or that option needs to be removed.

Reply:
I think we should merge this PR as it does not change how this code was working, and address this issue in another PR.

Reply:
I believe I added this to mimic the lxml etree API, see "Serialising to Unicode strings" paragraph in https://lxml.de/parsing.html

or corresponding code in lxml/etree.pyx:
 https://github.com/lxml/lxml/blob/71667f9ac7694216ee8e793192bcd0993a0cdc66/src/lxml/etree.pyx#L3413-L3418

but Just said it doesn't actually work, need to investigate why. let's file a separate issue/PR for that